tasks/src/main/java/ru/sberbank/school/task02/calculator/ExtendedCurrencyCalculator.java

```diff
@@ -0,0 +1,118 @@
+import ru.sberbank.school.task02.util.Symbol;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.math.RoundingMode;
+import java.util.stream.Collectors;
+
+public class ExtendedCurrencyCalculator extends CurrencyCalculator implements ExtendedFxConversionService {
+    private int rounding_mode;
+
+    public ExtendedCurrencyCalculator(ExternalQuotesService externalQuotesService) {
+        super(externalQuotesService);
+    }
+
+    private Optional<BigDecimal> extendedOperation(ClientOperation operation, Beneficiary beneficiary) {
+        Optional<Quote> quote = findQuote(new CompareQuotesByPrice(operation, beneficiary), quotes);
+        if (!quote.isPresent()) {
+            System.out.println("Return Optional.empty()");
+            return Optional.empty();
+        }
+        showQuote(quote.get());
+        if (operation == ClientOperation.SELL) {
+                System.out.println("Returl value: " + BigDecimal.valueOf(1).divide(quote.get().getBid(),10, rounding_mode));
+                return Optional.of(BigDecimal.valueOf(1).divide(quote.get().getBid(),10, RoundingMode.HALF_UP));
+        }
+        if (operation == ClientOperation.BUY) {
+            System.out.println("Returl value: " + BigDecimal.valueOf(1).divide(quote.get().getOffer(),10, rounding_mode));
+            return Optional.of(BigDecimal.valueOf(1).divide(quote.get().getOffer(), 10, RoundingMode.HALF_UP));
+        }
+        System.out.println("Return Optional.empty()");
+        return Optional.empty();
+    }
+
+    Optional<Quote> findQuote(Comparator<Quote> comparator, List<Quote> quotes) {
+        BigDecimal amountQuote;
+        List<Quote> finalQuoteList = new ArrayList<>();
+        for (Quote quote: quotes) {
+            if (quote.getVolume().isInfinity() || quote.getVolumeSize().compareTo(amountOfRequest) > 0) {
+                finalQuoteList.add(quote);
+            }
+        }
+
+        List<Quote> sortedQuoteList = filterQutesList(comparator, finalQuoteList);
+        for (Quote quote : sortedQuoteList) {
+            amountQuote = quote.getVolumeSize();
+            if (amountQuote.compareTo(amountOfRequest) > 0) {
+                return Optional.of(quote);
+            }
+            if (quote.getVolume().isInfinity()) {
+                return Optional.of(quote);
+            }
+        }
+        return Optional.empty();
+    }
+
+    List<Quote> filterQutesList(Comparator<Quote> comparator, List<Quote> quotes) {
+        List<Quote> filterBySymbolList = quotes.stream()
+                .sorted(comparator)
+                .collect(Collectors.toList());
+        System.out.println("Sorted List: ");
+        showQuotes(filterBySymbolList);
+        return filterBySymbolList;
+    }
+
+
+    @Override
+    public Optional<BigDecimal> convertReversed(ClientOperation operation, Symbol symbol, BigDecimal amount, Beneficiary beneficiary) {
```

Comment:
Line is longer than 120 characters (found 135).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1658568428>*